score-v1b1.json

```diff
@@ -144,11 +144,11 @@
           "pattern": "^[A-Za-z0-9][A-Za-z0-9-]{0,61}[A-Za-z0-9]$"
         },
         "id": {
-          "description": "An optional external Resource identifier. When two resources share the same type, class, and id, they are considered the same resource when used across related Workloads. The id must be a valid RFC1123 Label Name of up to 63 characters, including a-z, 0-9, '-' but may not start or end with '-'.",
+          "description": "An optional Resource identifier. When two resources share the same type, class, and id, they are considered the same resource when used across related Workloads. The id may be up to 63 characters, including a-z, 0-9, '-' or '.' but may not start or end with '-' or '.'.",
```

Comment:
Do we want to allow `..`, `.-` or `-.`? This would allow the IDs to also be valid DNS subdomains (i.e a series of DNS labels separated by `.`)

Given there is already `maxLength`, maybe we could reformulate the regex to:
`^[a-z0-9]+(?:-+[a-z0-9]+)*(?:\.[a-z0-9]+(?:-+[a-z0-9]+)*)*$`

This would allow things like:
`a.b, `a--b`, `a-b.c.d-e` but exclude `a-.b`, `-.-` `a.b.c-`

Reply:
Yes this is true, and probably more valid - I've tested the regex you suggested in regex101 and it seems to work well, so I've updated to that. 
I'm not super-keen on expanding this regex to be a full "valid a dns name" regex which can be super long and it might be more effective to provide a deeper validation function in the score-go library to do a more thorough approach with human-readable error messages.